spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.java

```diff
@@ -363,4 +392,51 @@ protected ElasticsearchHealthIndicator createHealthIndicator(Client client) {
 
 	}
 
+	/**
+	 * A {@link BeanPostProcessor} that registers {@link HealthIndicator} beans with a
+	 * {@link HealthIndicatorRegistry}.
+	 */
+	private static class HealthIndicatorRegistryPostProcessor implements BeanPostProcessor {
```

Comment:
Sorry, I still don't see the need for the post-processor. It feels more complex than the current code which is essentially doing the same thing